### PR TITLE
camera follow updates

### PR DESCRIPTION
**Pull request by tsboth**
It will need some changes after the car driving is implemented

### CODE REVIEW

File: Game/MyProject/Game/SceneManagement/SceneNode.cpp [hunk 1]
```diff
@@ -159,9 +173,23 @@ void SceneNode::setAnimator(const std::shared_ptr<NodeAnimator>& animator) {
...
     _rendarable = renderable;
 }
 
+void SceneNode::setIsMoving(bool value) {
+    _is_moving = value;
+}
+
+// returns true is the node is moving
+bool SceneNode::getIsMoving() {
+    return _is_moving;
```
Comment (raxvan) [inline Game/MyProject/Game/SceneManagement/SceneNode.cpp]: ideal case is to ask the animator if the node is moving with a virtual function for example. You can keep the bool value and assign it in node update function not via `setIsMoving` as an optimization.

**Review by raxvan — COMMENTED** [review]
small change.

File: Game/MyProject/Game/SceneManagement/CameraFollowAnimator.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#include "CameraFollowAnimator.hpp"
+
+#include "CameraSceneNode.hpp"
+#include "CarAnimator.hpp"
+
+CameraFollowAnimator::CameraFollowAnimator(const SceneNode* target, const IvVector2& camera_circle, bool needs_smoothing) :
+    _target(target), _camera_circle(camera_circle), _needs_smoothing(needs_smoothing) {
+        // setting default values
+        if (needs_smoothing) {
+            _min_distance = 10.;
```
Comment (mircea-catana) [inline Game/MyProject/Game/SceneManagement/CameraFollowAnimator.cpp]: Just a small note. I appreciate the fact that you try to write the correct data type so there is no conversion from int to float, but when you write "10." that's actually a double literal, so the conversion still happens double->float. The correct way to write float literals is either "10.f" or "10.0f"

**Review by mircea-catana — APPROVED** [review]
lgtm

**Pull request merged**